tzinfo/private/zoneinfo.rkt

```diff
@@ -117,12 +121,32 @@
         "/usr/share/lib/zoneinfo"
         "/etc/zoneinfo"))
 
-;; If the tzdata package (ver. >= 0.5) is installed, its zoneinfo
-;; directory will be here.
-(define-runtime-path tzdata-zoneinfo-path '(share "tzdata/zoneinfo"))
+;; The 'share option of define-runtime-path was added in version 7.5.0.7.
+(define-runtime-path-list relocatable-tzdata-zoneinfo-path-list
+  (cond
+    [(version<? (version) "7.5.0.7")
+     null]
+    [else
+     (list '(share "tzdata/zoneinfo"))]))
+
+;; For older versions of Racket, we can't correctly support
+;; `raco distribute`, since this package may have been built on a
+;; different system (with a different share directory path) than the one
+;; on which `raco distribute` is run. But we can support non-`raco
+;; distribute` uses.
+(define non-relocatable-tzdata-zoneinfo-path-list
+  (cond
+    [(version<? (version) "7.5.0.7")
+     (list (build-path (find-user-share-dir) "tzdata/zoneinfo")
+           (build-path (find-share-dir) "tzdata/zoneinfo"))]
+    [else
+     null]))
 
 (define current-zoneinfo-search-path
-  (make-parameter (cons tzdata-zoneinfo-path default-zoneinfo-search-path)))
+  (make-parameter
+   (append relocatable-tzdata-zoneinfo-path-list
+           non-relocatable-tzdata-zoneinfo-path-list
+           default-zoneinfo-search-path)))
```

Comment:
@mflatt I don't know if you recall the [discussion on racket-users from two years ago](https://groups.google.com/g/racket-users/c/_2lHI5SkqJo/m/9GZThigMCAAJ) that led to the addition of the `'share` mode to `define-runtime-path`, but I'm curious if you think this is a reasonable way to support versions of Racket that pre-date that change, as well as ones that have it.